src/math/lp/dioph_eq.cpp

```diff
@@ -0,0 +1,2458 @@
+               sum {ri*xi|i!= k} + c_r. Then -xt + x_k + sum {qi*x_i)| i != k} + c_q
+               will be the fresh row eh = ahk*xt +  sum {ri*x_i | i != k} + c_r  is
+               the row m_e_matrix[e.m_row_index]
+            */
+            mpq q, r;
+            lar_term fresh_t;  // fresh term
+            fresh_t.add_monomial(-mpq(1), xt);
+            fresh_t.add_monomial(mpq(1), k);
+            for (const auto& i : m_e_matrix.m_rows[h]) {
+                const mpq& ai = i.coeff();
+                if (i.var() == k)
+                    continue;
+                q = machine_div_rem(ai, ahk, r);
+                if (!q.is_zero())
+                    fresh_t.add_monomial(q, i.var());
+            }
+
+            m_fresh_k2xt_terms.add(k, xt, std::make_pair(fresh_t, h));
+            SASSERT(var_is_fresh(xt));
+            register_var_in_fresh_defs(h, xt);
+            eliminate_var_in_f_with_term(fresh_t, k, 1);
+        }
+
+        std::ostream& print_entry(unsigned i, std::ostream& out,
+                                  bool print_dep = false) {
+            out << "entry " << i << ":";
+            out << "{\n";
+            print_term_o(get_term_from_entry(i), out << "\tm_e:") << ",\n";
+            // out << "\tstatus:" << (int)e.m_entry_status;
+            if (print_dep) {
+                auto l_term = l_term_from_row(i);
+                out << "\tm_l:{";
+                print_lar_term_L(l_term, out) << "}, ";
+                print_ml(l_term, out) << std::endl;
+                out << "expl of fixed in m_l:{\n";
+                print_deps(out, explain_fixed_in_meta_term(l_term));
+                out << "}\n";
+            }
+            if (belongs_to_f(i)) {
+                out << "in F\n";
+            } else {
+                unsigned j = m_k2s.get_key(i);
+                if (local_to_lar_solver(j) == UINT_MAX) {
+                    out << "FRESH\n";
+                } else {
+                    out << "in S\n";
+                }
+            }
+            out << "}\n";
+            return out;
+        }
+
+        bool belongs_to_f(unsigned ei) const {
+            return !m_k2s.has_val(ei);
+        }
+
+        void remove_from_S(unsigned ei) {
+            m_k2s.erase_val(ei);
+        }
+
+        // k is the variables that is being substituted
+        // h is the index of the entry
+        void move_entry_from_f_to_s(unsigned k, unsigned h) {
+            m_k2s.add(k, h);
+        }
+
+        // this is the step 6 or 7 of the algorithm
+        // returns true if an equlity was rewritten and false otherwise
+        bool rewrite_eqs(std_vector<unsigned>& f_vector) {
+            if (f_vector.size() == 0)
+                return false;
+            unsigned h = -1;
+            unsigned n = 0;  // number of choices for a fresh variable
+            mpq min_ahk;
+            unsigned kh;
+            int kh_sign;
+            unsigned h_markovich_number;
```

Comment:
compiler warns this h_markovich_number may be uninitialized, 
Add some default initial values to appropriate variables, or declare them in place they are assigned.